### ISSUE TITLE
"Made for you" - "Discover Weekly" seems to play the wrong song when first selecting a song

### ISSUE THREAD

**Issue by Bios-Marcel** [Rigellute/spotify-tui#313]
When trying to play a song from "Discover Weekly" (selected in the image), it just plays something else that is seemingly random. Maybe the view and the model are somehow desynchronized? Like it still has some old playlist in the background. I can't really figure out what comes from where.

![grafik](https://user-images.githubusercontent.com/19377618/75337796-f0b13600-588d-11ea-84bc-6bebecafd531.png)

**Comment by Rigellute** [comment]
Just gave this a try and it seems to be working normally for me. 

Were there any additional steps/actions before you got to the "Discover Weekly" that would help me reproduce?

**Comment by Bios-Marcel** [comment]
Today I can't manage to reproduce it anymore ... weird. If I encounter this again, I'll try giving additional information. Maybe this isn't even related to "Discover Weekly". I assume it's some kind of general problem.

**Comment by isovector** [comment]
I think I've run into this; if I press enter on a playlist and then scroll to a different playlist before the network request finishes, it seems to throw off spt's state.

**Comment by ericonr** [comment]
> I think I've run into this; if I press enter on a playlist and then scroll to a different playlist before the network request finishes, it seems to throw off spt's state.

Exactly. I've had this happen where it plays songs from Release Radar instead of Discover Weekly, and going up or down in the song list chooses different songs from Release Radar, even though I see the song list for Discover Weekly.

And it's probably more apparent on slow connections.
